src/emitter.js

```diff
@@ -0,0 +1,279 @@
+
+			// Finally register the callback to the event.
+			emitter.on( event, callback, ctx, priority );
+		},
+
+		/**
+		 * Stops listening for events. It can be usued at different levels:
+		 *
+		 *  * To stop listening to a specific callback.
+		 *  * To stop listening to a specific event.
+		 *  * To stop listening to all events fired by a specific object.
+		 *  * To stop listening to all events fired by all object.
+		 *
+		 * @param {Emitter} [emitter] The object to stop listening to. If omitted, stops it for all objects.
+		 * @param {String} [event] (Requires `emitter`) The name of the event to stop listening to. If omitted, stops it
+		 * for all events from `emitter`.
+		 * @param {Function} callback (Requires `event`) The function be removed from the call list for the give
+		 * `event`.
+		 */
+		stopListening: function( emitter, event, callback ) {
+			var emitters = this._listeningTo;
+			var emitterId = emitter && emitter._emitterId;
+			var emitterInfo = emitters && emitterId && emitters[ emitterId ];
+			var eventCallbacks = emitterInfo && event && emitterInfo.callbacks[ event ];
+
+			// Stop if nothing has been listened.
+			if ( !emitters || ( emitter && !emitterInfo ) || ( event && !eventCallbacks ) ) {
+				return;
+			}
+
+			if ( callback ) {
+				// All params provided. off() that single callback.
+				emitter.off( event, callback );
+			} else if ( eventCallbacks ) {
+				// Only emitter and event provided. off() all callbacks for that event.
+				while ( ( callback = eventCallbacks.pop() ) ) {
+					emitter.off( event, callback );
+				}
+				delete emitterInfo.callbacks[ event ];
+			} else if ( emitterInfo ) {
+				// Only emitter provided. off() all events for that emitter.
+				for ( event in emitterInfo.callbacks ) {
+					this.stopListening( emitter, event );
+				}
+				delete emitters[ emitterId ];
+			} else {
+				// No params provided. off() all emitters.
+				for ( emitterId in emitters ) {
+					this.stopListening( emitters[ emitterId ].emitter );
+				}
+				delete this._listeningTo;
+			}
+		},
+
+		/**
+		 * Fires and event, executing all callbacks registered for it.
+		 *
+		 * The first parameter passed to callbacks is a {EventInfo} object, followed by the optional `args` provided in
+		 * the `fire()` method call.
+		 *
+		 * @param {String} event The name of the event.
+		 * @param {...*} [args] Additional arguments to be passed to the callbacks.
+		 */
+		fire: function( event, args ) {
+			var callbacks = getCallbacksIfAny( this, event );
+
+			if ( !callbacks ) {
+				return;
+			}
+
+			var eventInfo = new EventInfo( event );
+
+			// Take the list of arguments to pass to the callbacks.
+			args = Array.prototype.slice.call( arguments, 1 );
+			args.unshift( eventInfo );
+
+			for ( var i = 0; i < callbacks.length; i++ ) {
+				callbacks[ i ].callback.apply( callbacks[ i ].ctx, args );
+
+				if ( eventInfo.stop.called ) {
+					break;
+				}
+
+				if ( eventInfo.off.called ) {
```

Comment:
Thing that worries me is that if `stop()` was called, then `off()` call will be ignored. It's not what I would expect.